### PR TITLE
feat(ci): Add initial test case for a Docker build in CI for multi-platform builds

### PR DESCRIPTION
**Pull request by jacobdr**
## Change Summary

This PR adds a basic docker build setup that exercises the `prisma` CLI, building for multiple platforms in Docker via `--platform=linux/amd64,linux=arm64`

Additionally, it starts to exercise the build against multiple [Linux distributions for Python](https://hub.docker.com/_/python), starting with a basic support matrix for Debian Buster and Alpine (latest).

The scope is **not to publish** any sort of official image, but rather since GH Actions does not support an ARM CPU runner we can emulate how a build would have performed via the QEMU emulation of docker.

Relates to #454
Relates to #461
Relates to #507 

CI results for this running off of my fork can be found here: https://github.com/jacobdr/prisma-client-py/pull/1

## Checklist

- [ ] Unit tests for the changes exist
- [ ] Tests pass without significant drop in coverage
- [ ] Documentation reflects changes where applicable
- [ ] Test snapshots have been [updated](https://prisma-client-py.readthedocs.io/en/latest/contributing/contributing/#snapshot-tests) if applicable

## Agreement

By submitting this pull request, I confirm that you can use, modify, copy and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by RobertCraigie** [comment]
Love this change, I should've done this ages ago, thanks!

This looks good to me but I'd prefer if it was more obvious that the `Dockerfile` is only used for testing purposes. I'm not sure where it would best fit into the current architecture though...

**Comment by jacobdr** [comment]
I could place it in the `tests` directory so its a bit more hidden and leave a comment up at the top of the file indicating why. Would that be sufficient?

File: .github/workflows/test.yml [hunk 1]
```diff
@@ -248,3 +248,31 @@ jobs:
...
+  docker:
+    name: docker
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        docker-platform: ["linux/amd64"]
+        # TODO: Uncomment this to add testing support for arm64 and delete
+        # the above
+        # docker-platform: ["linux/amd64", "linux/arm64"]
```
Comment (RobertCraigie) [inline .github/workflows/test.yml:261]: Is `linux/arm64` currently broken?

Comment (RobertCraigie) [reply .github/workflows/test.yml:261]: Just saw your comment in another file, what was the error?

**Review by RobertCraigie — APPROVED** [review]
All looks good to me, thanks!

**Pull request merged**